ds0_updater.py

```diff
@@ -5,13 +5,36 @@
 import logger
 import conf
 
+# keep it super simple for now because we don't really implement --allowed-keys yet.
+def verify_latest_commit():
+    allowed_keys = ds0_utils.allowed_keys()
+
+    # ensure the latest commit is signed and verified
+    while True:
+        for key in allowed_keys:
+            logger.exe('gpg --import ' + key['file'], expectError=True)
+        output = logger.exe('git log --pretty="format:%G?" --show-signature HEAD^..HEAD')
+
+        if "Can't check signature" in output[0]:
+            logger.info('gpg keys cleared on startup. Trying again...')
+            continue
+
+        verified = False
+        for key in allowed_keys:
+            if key['rsa_check'] in output[0] and key['signature_check'] in output[0]:
+                verified = True
+        if not verified:
+            logger.error('Scripts using a non-signed commit. Please ensure commit is valid.')
+            logger.error('    If it was a missed signature, feel free to open a ticket at https://github.com/riptano/ComboAMI.')
+        break
+
 # Update the AMI codebase if it's its first boot
 if not conf.get_config("AMI", "CompletedFirstBoot"):
     # check if a specific commit was requested
     force_commit = ds0_utils.required_commit()
 
     # update the repo
-    logger.exe('git pull')
+    logger.exe('git pull ' + ds0_utils.repository())
```

Comment:
Unless we do some rejiggering from line 37 to 44, we have something less flexible than what you started with. I'm actually fine with this, though... it means:

1. --force_commit only works for commits that are in the official riptano/ComboAMI repository... and so is mainly useful for project commiters to test things.
2. --repository fast-forwards to the latest commit in the 2.5 branch of a user-specified repo. It's mainly useful to community members that want to tweak the AMI startup-scripts in a lightweight way. They can't specify an arbitrary commit or branch, and must push their changes to fork with a branch named '2.5'.

While this isn't arbitrarily flexible, it seems to me that it's still quite useful. I'm fine with this unless you want to rethink the git strategy. I don't have a better suggestion off the top of my head.